code/app/src/main/java/com/adobe/marketing/mobile/messagingsample/MainActivity.kt

```diff
@@ -65,15 +335,125 @@ class MainActivity : AppCompatActivity() {
         val actionReceiver = Intent(applicationContext, MainActivity::class.java).apply {
             flags = Intent.FLAG_ACTIVITY_NEW_TASK or Intent.FLAG_ACTIVITY_CLEAR_TASK
             putExtra(FROM, "action")
-            Messaging.addPushTrackingDetails(this, "messageId", NotificationBroadcastReceiver.XDM_DATA)
+            Messaging.addPushTrackingDetails(
+                this,
+                "messageId",
+                NotificationBroadcastReceiver.XDM_DATA
+            )
         }
-        val pendingIntent: PendingIntent = PendingIntent.getBroadcast(this, System.currentTimeMillis().toInt(), actionReceiver, 0)
-        builder.addAction(R.drawable.ic_launcher_background, "buttonAction",
-                pendingIntent)
+        val pendingIntent: PendingIntent =
+            PendingIntent.getBroadcast(this, System.currentTimeMillis().toInt(), actionReceiver, 0)
+        builder.addAction(
+            R.drawable.ic_launcher_background, "buttonAction",
+            pendingIntent
+        )
         return builder.build()
     }
 
     companion object {
         const val FROM = "from"
     }
+
+    override fun onResume() {
+        super.onResume()
+        MobileCore.lifecycleStart(null)
```

Comment:
Lets do this in onStart

Reply:
we recommend calling lifecycleStart from onResume as onStart is not called when a user returns to an activity.